File: demo-service-impl/src/main/java/com/github/zubmike/service/demo/dao/PlanetarySystemDao.java
package com.github.zubmike.service.demo.dao;

import com.github.zubmike.service.dao.DictItemDao;
import com.github.zubmike.service.demo.types.PlanetarySystem;

import java.util.Optional;

public interface PlanetarySystemDao extends DictItemDao<Integer, PlanetarySystem> {

	Optional<PlanetarySystem> getByCode(String code);

}

File: demo-service-kotlin-impl/src/main/kotlin/com/github/zubmike/service/demo/dao/PlanetarySystemDao.kt
package com.github.zubmike.service.demo.dao

import com.github.zubmike.core.dao.DictItemDao
import com.github.zubmike.service.demo.types.PlanetarySystem

interface PlanetarySystemDao : DictItemDao<Int, PlanetarySystem> {

    fun getByCode(code: String): PlanetarySystem?

}
